File: POO JAVA/Aula09/src/aula09/Livro.java
package aula09;

import java.util.Random;

public class Livro implements Publicacao {

    //Atributos da Classe
    private String tiluto;
    private String autor;
    private int totPaginas;
    private int pagAtual;
    private boolean aberta;
    private Pessoa leitor;
    
    //Metodo Construtor
    public Livro(String titul, String aut, int totPag, Pessoa leitor) {
        this.tiluto = titul;
        this.autor = aut;
        this.pagAtual = 0;
        this.aberta = false;
        this.totPaginas = totPag;
        this.leitor = leitor;
    }
    
    //Metodos Acessores e Modificadores da Classe
    public String getTiluto() {
        return tiluto;
    }
    public void setTiluto(String tiluto) {
        this.tiluto = tiluto;
    }
    public String getAutor() {
        return autor;
    }
    public void setAutor(String autor) {
        this.autor = autor;
    }
    public int getTotPaginas() {
        return totPaginas;
    }
    public void setTotPaginas(int totPaginas) {
        this.totPaginas = totPaginas;
    }
    public int getPagAtual() {
        return pagAtual;
    }
    public void setPagAtual(int pagAtual) {
        this.pagAtual = pagAtual;
    }
    public boolean isAberta() {
        return aberta;
    }
    public void setAberta(boolean aberta) {
        this.aberta = aberta;
    }
    public Pessoa getLeitor() {
        return leitor;
    }
    public void setLeitor(Pessoa leitor) {
        this.leitor = leitor;
    }
    
    //Metodo da Classe
    public String detalhes() {
        return "Livro \n{" + "\nTiluto = " + tiluto + ", \nAutor = " + autor + ","
                + " \nTotal de Paginas = " + totPaginas + ", \nPagina Atual = " + pagAtual + ","
                + " \nAberta = " + aberta + ", \nLeitor = " + leitor.getNome() + "\nIdade = "
                + leitor.getIdade() + "\nSexo = " + leitor.getSexo() + "\n}";
    }
    
    
    //Metodos da Interface
    @Override
    public void abrir() {
        this.aberta = true;
    }
    @Override
    public void fechar() {
        this.aberta = false;
    }
    @Override
    public void folhear(int folhear){
        if((this.getTotPaginas() >= folhear) && (this.getTotPaginas() >= this.getPagAtual()+folhear)){
            this.pagAtual = folhear;
        }else{
            System.out.println("Erro. Pagina do livro insuficiente.");
        }
    }
    @Override
    public void avançarPag() {
        this.pagAtual++;
    }
    @Override
    public void voltarPag() {
        this.pagAtual--;
    }
    
}

File: POO PHP/Aula09/Livro.php
<?php
require_once 'Pessoa.php';
require_once 'Publicacao.php';

class Livro implements Publicacao{

    //Atributos da Classe
    private $titulo;
    private $autor;
    private $totPaginas;
    private $pagAtual;
    private $aberto;
    private $leitor;
    
    //Metodo Construtor
    public function __construct($titulo, $autor, $totPaginas, $leitor) {
        $this->titulo = $titulo;
        $this->autor = $autor;
        $this->totPaginas = $totPaginas;
        $this->pagAtual = 0;
        $this->aberto = false;
        $this->leitor = $leitor;
    }

    //Metodos Acessores e Modificadores
    public function getTitulo() {
        return $this->titulo;
    }
    public function getAutor() {
        return $this->autor;
    }
    public function getTotPaginas() {
        return $this->totPaginas;
    }
    public function getPagAtual() {
        return $this->pagAtual;
    }
    public function getAberto() {
        return $this->aberto;
    }
    public function getLeitor(): Pessoa {
        return $this->leitor;
    }
    public function setTitulo($titulo): void {
        $this->titulo = $titulo;
    }
    public function setAutor($autor): void {
        $this->autor = $autor;
    }
    public function setTotPaginas($totPaginas): void {
        $this->totPaginas = $totPaginas;
    }
    public function setPagAtual($pagAtual): void {
        $this->pagAtual = $pagAtual;
    }
    public function setAberto($aberto): void {
        $this->aberto = $aberto;
    }
    public function setLeitor(Pessoa $leitor): void {
        $this->leitor = $leitor;
    }

    //Metodos da Classe
    public function detalhes() {
        echo "<br><br>====  Livro  ====";
        echo "<br>Titulo: ".$this->titulo;
        echo "<br>Autor: ".$this->autor;
        echo "<br>Total de Paginas: ".$this->totPaginas;
        echo "<br>Pagina Atual: ".$this->pagAtual;
        echo "<br>Livro Aberto: ".$this->aberto;
        echo "<br>Leitor: ".$this->leitor->getNome();
        echo "<br>Idade: ".$this->leitor->getIdade();
        echo "<br>Sexo: ".$this->leitor->getSexo();
    }

    //Metodos Implementados
    public function abrir() {
        $this->aberto = true;
    }
    public function avannçarPagina() {
        if($this->pagAtual++ > $this->totPaginas){
            echo "Erro! Fim de paginas do livro";
        } else {
            $this->pagAtual++;
        }
    }
    public function fechar() {
        $this->aberto = false;
    }
    public function folhear($pag) {
        if ($pag > $this->getTotPaginas()){
            $this->pagAtual = 0;
            echo "Erro! Paginas do livro insuficiente";
        } else {
            $this->setPagAtual($pag);
        }
    }
    public function voltarPagina() {
        if($this->pagAtual-- < 1){
            echo "Erro! Livro Fechado";
        } else {
            $this->pagAtual--;
        }
    }
    
}
